app/src/main/java/com/duckduckgo/app/global/rating/ShowInitialPromptDecider.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.global.rating
+
+import com.duckduckgo.app.browser.rating.db.AppEnjoymentRepository
+import com.duckduckgo.app.usage.app.AppDaysUsedRepository
+import timber.log.Timber
+
+
+interface ShowPromptDecider {
+    suspend fun shouldShowPrompt(): Boolean
+}
+
+class InitialPromptDecider(
+    private val appDaysUsedRepository: AppDaysUsedRepository,
+    private val appEnjoymentRepository: AppEnjoymentRepository
+) : ShowPromptDecider {
+
+    override suspend fun shouldShowPrompt(): Boolean {
+        if (!enoughDaysPassedToShowFirstPrompt()) {
+            Timber.i("Not enough days passed to show first prompt")
+            return false
+        }
+
+        if (appEnjoymentRepository.hasUserPreviouslySeenFirstPrompt()) {
+            Timber.i("User has seen first prompt already")
+            return false
+        }
+
+        return true
+    }
+
+    private suspend fun enoughDaysPassedToShowFirstPrompt(): Boolean {
+        val daysUsed = appDaysUsedRepository.getNumberOfDaysAppUsed()
+        val enoughDaysUsed = daysUsed >= MINIMUM_DAYS_USAGE_BEFORE_FIRST_PROMPT
+
+        return enoughDaysUsed.also {
+            Timber.i("Number of days usage: $daysUsed. Enough days have passed to show app enjoyment prompt: %s", if (enoughDaysUsed) "yes" else "no")
```

Comment:
>Is this a new kotlin style thing

No; I don't think so. There does happen to an example of using `.also` for a log statement on the kotlinlang site[1], but I don't think that's a compelling reason to use it per se.

I had liked the way it relegated the log statement to a sort of side-effect; something not important enough to be in the main block where the real logic was. i.e., i hoped it would _denoise_ the logging from the logic. I can see that attempt failed!

Happy to go with more traditional style here if that's clearer.

[1]
https://kotlinlang.org/docs/reference/coding-conventions.html#using-scope-functions-applywithrunalsolet